### PR TITLE
Use image server for all kind of images and integrate identity rings 

### LINKED ISSUE #13670: Use image server for all kind of images and integrate identity rings
### Description

Currently the http server in status-go https://github.com/status-im/status-go/blob/develop/server/server.go is used only for identicon and image messages.
We should also use it for profile pictures. 

There's a branch out that also handles this case, which includes work from @qfrank:
https://github.com/status-im/status-mobile/tree/feature/identity-images-http-server
https://github.com/status-im/status-go/tree/feature/13334

This branch has a few changes:

- Adds an identity ring to the image
- Adds a signal when the port of the http server changes
- Serves user profile pictures

Some of the work integrating the branch has already been done here:

The important things to note when finishing integrating the branch:

- If the port of the server changes, the URL should of the image should be updated and re-fetched
- Status-go should not serve user images if the user has their settings set to not display profile pictures, only identicons should be server
- If this setting changes, pictures should be reloaded
- Dark/light theme should be likely passed as a parameter, so that the identity ring can be correctly set
- Identity ring should be toggled, as we don't want to show it in the old UI, only on the new UI

### PR DESCRIPTION
**Pull request by qfrank**
fixes #13670

### CODE REVIEW

File: src/status_im/subs.cljs [hunk 1]
```diff
@@ -2268,13 +2289,35 @@
  (fn [multiaccount]
    (get multiaccount :profile-pictures-visibility)))
 
+(defn- replace-contact-image-uri [contact port identity]
+  (let [theme (colors/current-theme)
+        identicon (image-server/get-identicons-uri port identity theme)
+        contact-images (:images contact)
+        contact-images (reduce (fn [acc image] (let [image-name (:type image)
+                                                       ; We pass the clock so that we reload the image if the image is updated
```
Comment (J-Son89) [inline src/status_im/subs.cljs:2343]: This is some form of hack? 🤔 if so, is there no cleaner way of reloading the image on updates?

File: src/status_im/signals/core.cljs [hunk 2]
```diff
@@ -70,6 +70,7 @@
       "mailserver.available" (mailserver/handle-mailserver-available cofx (.-id event-js))
       "mailserver.not.working" (mailserver/handle-mailserver-not-working cofx)
       "discovery.summary"  (summary cofx (js->clj event-js :keywordize-keys true))
+      "mediaserver.started"  {:db (assoc db :mediaserver/port (.-port event-js))}
```
Comment (flexsurfer) [inline src/status_im/signals/core.cljs:73]: why do we need a separate signal? is it possible to start it on sign-in?

File: src/status_im/subs.cljs [hunk 3]
```diff
@@ -2953,6 +3002,27 @@
  (fn [multiaccount]
    (pos? (count (get multiaccount :images)))))
 
+(re-frame/reg-sub
+ :profile/multiaccount
```
Comment (J-Son89) [inline src/status_im/subs.cljs:3066]: nit: should the variable name be `multi-account`?

File: src/quo/design_system/colors.cljs [hunk 4]
```diff
@@ -191,6 +191,12 @@
 
 (def theme-type (reagent/atom :light))
 
+(defn current-theme []
```
Comment (flexsurfer) [inline src/quo/design_system/colors.cljs]: mhm, probably its only related to image server better to move this to image server ns

File: src/status_im/subs.cljs (hunk 3 above)
Comment (qfrank) [reply src/status_im/subs.cljs:3066]: i wish name it multi-account too...but the old code called it multiaccount, so i just want keep the same :)

File: src/quo/design_system/colors.cljs (hunk 4 above)
Comment (flexsurfer) [reply src/quo/design_system/colors.cljs]: or rename it to `current-theme-int`

File: src/status_im/subs.cljs [hunk 5]
```diff
@@ -1282,15 +1285,33 @@
...
  :<- [:contacts/contacts]
+ :<- [:profile/multiaccount]
+ :<- [:mediaserver/port]
+ (fn [[contacts multiaccount port] [_ id identicon]]
+   (let [contact (or (get contacts id)
+                     (when (= id (:public-key multiaccount))
+                       multiaccount))]
+     (if-not (nil? contact)
+       (multiaccounts/displayed-photo contact)
+       (if @config/new-ui-enabled?
```
Comment (flexsurfer) [inline src/status_im/subs.cljs]: why only for new design?

File: src/status_im/subs.cljs (hunk 1 above)
Comment (qfrank) [reply src/status_im/subs.cljs:2343]: hi @J-Son89  the "cleaner" way would be generate the image link on the status-go side, but there is some special params like "current theme / new-ui-enabled?" status-go don't know , if u have better idea, i'd like to hear :)

File: src/status_im/subs.cljs (hunk 5 above)
Comment (qfrank) [reply src/status_im/subs.cljs]: oh sorry, i think i misunderstood "Identity ring should be toggled, as we don't want to show it in the old UI, only on the new UI", when new-ui not enabled, we should also use image server but no ring, thanks for point out! @flexsurfer

File: src/status_im/signals/core.cljs (hunk 2 above)
Comment (qfrank) [reply src/status_im/signals/core.cljs:73]: @cammellos 

> why do we need a separate signal? is it possible to start it on sign-in?

Comment (cammellos) [reply src/status_im/signals/core.cljs:73]: @flexsurfer we do, but sometimes it gets killed and the port changes, in that case we need to notify the UI so it stops using that port and reload images which might have not loaded, we could rename it something better I suppose?

File: src/status_im/utils/image_server.cljs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  (:require [quo.design-system.colors :as colors]
+            [status-im.utils.config :as config]))
+
+(def ^:const image-server-uri-prefix "https://localhost:")
+(def ^:const identicons-action "/messages/identicons")
+(def ^:const account-images-action "/accountImages")
+(def ^:const contact-images-action "/contactImages")
+
+(defn current-theme []
+  (case @colors/theme-type
```
Comment (flexsurfer) [inline src/status_im/utils/image_server.cljs:11]: i would abstract image server from colors , and pass it as a parameter , but its up to you

**Review by flexsurfer — APPROVED** [review]
great job!

File: src/status_im/utils/image_server.cljs (hunk 6 above)
Comment (qfrank) [reply src/status_im/utils/image_server.cljs:11]: emm.. i thought image-server would be much easier to use without passing param "theme"

**Comment by churik** [comment]
@qfrank sorry for the delay, as we're close to the 1.20 release cut and this feature shouldn't be there.
When we cut the branch, I'll ask you to rebase all for testing (your forked repos in `status-go` and `status-react` side and corresponding branches as well)

**Comment by qfrank** [comment]
@churik okay :)

**Comment by qfrank** [comment]
Hi @churik , this PR rebased :)

**Comment by churik** [comment]
@qfrank 
### ISSUE 1: There are no border at all for all users in chats, user profile, contacts list etc. in both dark and light mode

photo_2022-09-12 12 36 34 jpeg 2022-09-12 12-37-12

**OS:** IOS, Android


### ISSUE 2: Can't see an updated image in my profile until re-login 

**Steps:**
1) restore multi-account from seed phrase
2) open Profile > Edit image > Set new image


**Expected result:**
can see updated image

**Actual result:**

https://user-images.githubusercontent.com/4557972/189636783-0953a0b6-c3c7-4589-9b73-727d3506b144.mp4




**OS:** IOS, Android

**Comment by qfrank** [comment]
Hi @churik , the ring only available when new-ui enabled

**Comment by qfrank** [comment]
Hi @churik , the ring only available when new-ui enabled

**Comment by churik** [comment]
@qfrank yep, I got it, issues are regression for current UI
Please, ping me when it is ready to be retested.
Thank you!

**Comment by flexsurfer** [comment]
@qfrank so i guess for issue1 we lost a border

**Comment by qfrank** [comment]
> @qfrank yep, I got it, issues are regression for current UI Please, ping me when it is ready to be retested. Thank you!

sure

**Comment by qfrank** [comment]
Hi @churik , ready to be retested :)

**Comment by churik** [comment]
No regression anymore, thank you for fixes @qfrank 
e2e failures are not related to PR.
Tested basic flows, LGTM!

**Devices:**
- IPhone 12 Mini (IOS 15)
- Xiaomi Mi Note 9 Pro (Android 10)

**Pull request merged**